api/src/main/java/org/openmrs/module/reporting/config/ReportLoader.java

```diff
@@ -0,0 +1,313 @@
+            }
+        }
+        return parameters;
+    }
+
+    public static Class<?> getParameterClass(String clazz) throws ClassNotFoundException {
+        if (clazz.equalsIgnoreCase("location")) {
+            return Location.class;
+        }
+        else if (clazz.equalsIgnoreCase("date")) {
+            return Date.class;
+        }
+        else {
+            return Context.loadClass(clazz);
+        }
+    }
+
+    public static Map<String, Object> constructMappings(List<Parameter> parameters) {
+        Map<String,Object> mappings = new HashMap<String, Object>();
+        for (Parameter parameter : parameters) {
+            mappings.put(parameter.getName(), "${" + parameter.getName() + "}");
+        }
+        return mappings;
+    }
+
+    public static DataSetDefinition constructDataSetDefinition(DataSetDescriptor dataSetDescriptor, File path, List<Parameter> parameters) {
+
+        DataSetDefinition dsd = null;
+
+        if ("sql".equalsIgnoreCase(dataSetDescriptor.getType())) {
+            dsd = constructSQLFileDataSetDefinition(dataSetDescriptor, path);
+        }
+        else {
+            throw new RuntimeException("Unsupported data set descriptor type: " + dataSetDescriptor.getType());
+        }
+
+        if (parameters != null) {
+            for (Parameter parameter : parameters) {
+                dsd.addParameter(parameter);
+            }
+        }
+
+        return dsd;
+    }
+
+    public static SqlFileDataSetDefinition constructSQLFileDataSetDefinition(DataSetDescriptor dataSetDescriptor, File path) {
+
+        SqlFileDataSetDefinition dsd = new SqlFileDataSetDefinition();
+
+        File sqlFile = new File(path, dataSetDescriptor.getConfig());
+
+        if (sqlFile.exists()) {
+            dsd.setSqlFile(sqlFile.getAbsolutePath());
+        }
+        else {
+            throw new RuntimeException("SQL file " + dataSetDescriptor.getConfig() + " not found");
+        }
+
+        return dsd;
+    }
+
+    public static List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition, ReportDescriptor reportDescriptor) {
+
+        List<ReportDesign> reportDesigns = new ArrayList<ReportDesign>();
+
+        // always do a default CSV design
+        if (reportDescriptor.getDesigns() == null || reportDescriptor.getDesigns().size() == 0) {
+            DesignDescriptor defaultDesignDecsriptor = new DesignDescriptor();
+            defaultDesignDecsriptor.setType("csv");
+            reportDescriptor.setDesigns(Collections.singletonList(defaultDesignDecsriptor));
+        }
+
+        for (DesignDescriptor designDescriptor : reportDescriptor.getDesigns()) {
+            ReportDesign design = null;
+            if (designDescriptor.getType().equalsIgnoreCase("csv")) {
+                design = constructCSVReportDesign(reportDefinition);
+            }
+            else if (designDescriptor.getType().equalsIgnoreCase("excel") || designDescriptor.getType().equalsIgnoreCase("xls")) {
```

Comment:
I supported two different names for Excel reports, if you think we want to be stricter and only support one or the other, lmk...